core/domain/voiceover_domain.py

```diff
@@ -60,39 +63,44 @@ def __init__(
         entity_type: str,
         entity_version: int,
         language_accent_code: str,
-        voiceovers: Dict[str, Dict[str, state_domain.Voiceover]]
+        voiceovers: Dict[str, Dict[
+            feconf.VoiceoverType, state_domain.Voiceover]]
     ) -> None:
-        """Constructs an EntityVoiceover domain object.
+        """Constructs an EntityVoiceovers domain object.
 
         Args:
             entity_id: str. The ID of the entity.
             entity_type: str. The type of the entity.
             entity_version: int. The version of the entity.
-            language_accent_code: str. The langauge accent code of the
+            language_accent_code: str. The language accent code of the
                 given voiceover.
             voiceovers: dict(str, dict(str, VoiceoverDict)). A dict representing
-                content-id as keys and a nested dict as values. The nested dict
-                contains 'manual' and 'auto' as keys and dict(Voiceover)
-                as values.
+                content IDs as keys and a nested dict as values. Each
+                nested dict contains VoiceoverType as keys and
+                VoiceoverDict as values.
         """
         self.entity_id = entity_id
         self.entity_type = entity_type
         self.entity_version = entity_version
         self.language_accent_code = language_accent_code
         self.voiceovers = voiceovers
 
-    def to_dict(self) -> EntityVoiceoverDict:
-        """Returns the dict representation of the EntityVoiceover object.
+    def to_dict(self) -> EntityVoiceoversDict:
+        """Returns the dict representation of the EntityVoiceovers object.
 
         Returns:
-            EntityVoiceoverDict. The dict representation of the
-            EntityVoiceover object.
+            EntityVoiceoversDict. The dict representation of the
+            EntityVoiceovers object.
         """
         voiceovers_dict = {}
         for content_id, voiceover_type_to_voiceover in self.voiceovers.items():
             voiceovers_dict[content_id] = {
-                'manual': voiceover_type_to_voiceover['manual'].to_dict(),
-                'auto': voiceover_type_to_voiceover['auto'].to_dict()
+                feconf.VoiceoverType.MANUAL.value: (
```

Comment:
Maybe you can use a dict comprehension here in case we add a new voiceover type, e.g. {voiceover_type.value: ... for voiceover_type in feconf.VoiceoverType.keys()} or something like that.

Reply:
Thank you for the guidance, as per our current scope we are not planning to introduce further voiceover types apart from the current ones i.e., 'manual' and 'auto'. I am leaving this unaddressed for now, if we plan to extend the scope further then we can definitely opt for this in the future.
WDYT?
Thank you.

Reply:
@Nik-09 This isn't really about extending scope, it's about keeping the codebase maintainable so that a change can be done seamlessly without updating a lot of stuff all over the codebase in future. I think you'll want to keep it explicit only if it is important, in that function, to be explicit about what the API is (e.g. maybe when controlling what gets passed to the frontend, say). But if it's a straightforward conversion of one object to another then I recommend a dict comprehension so it's a bit more agnostic.

So I think the question is whether it's important for the 'manual' and 'auto' to be explicitly specified here, and if so, why.

Reply:
Updated here & in the remaining sections. This is indeed helpful.
Thanks